### ISSUE TITLE
Taught this Lesson?  Let us know!

### ISSUE THREAD

**Issue by sstevens2** [carpentries-incubator/docker-introduction#23]
Hi all!  Please let us know if you are teaching this lesson by copying the info below and replying to this issue.

> Name:
> Institution:
> Dates Taught:
> Was this taught as a stand alone workshop or as part of another workshop / conference?
> What feedback do you have on the lesson?

**Comment by ChristinaLK** [comment]
Name: Christina Koch
Institution: UW Madison
Date Taught: 2020-12-09
Was this taught as a stand alone workshop or as part of another workshop / conference? stand alone
What feedback do you have on the lesson? taught mostly as written. can add notes to the instructor notes about how I execute certain parts of the lesson. Might switch the Docker hub / cleaning up container episodes (order wise). Have an image to add and a wrap-up exercise
Anything else you'd like to share?

**Comment by mkuzak** [comment]
Name: Mateusz Kuzak
Institution: The Netherlands eScience Center
Date Taught: 2021-03-23
https://esciencecenter-digital-skills.github.io/2021-03-23-containers/
Was this taught as a stand alone workshop or as part of another workshop / conference? stand alone
What feedback do you have on the lesson? 
Most of the things we shared elsewhere. One thing that was not captured anywhere is the analogy we used was that Dockerfies are like a design for the 3D printed cookie cutter, the 3D printed cookie cutter is the image and the cookie is the running container. Djura Smits made a graphic for it.
![docker-cookie-cutter](https://user-images.githubusercontent.com/208443/128470197-db279b7b-5130-4cdd-a7bb-d17929cec605.png)

**Comment by sstevens2** [comment]
We are hoping to get this lesson into the Carpentries Lesson review process in the next couple weeks. You are mentioned in this comment because you filed an issue that suggests you may have taught the lesson in the past.  As part of the review process we need to show that this lesson is teachable by individuals not involved in its design.  If you have taught it please reply to this thread and and let us know so we have a record of others having taught this lesson.  Thanks for your help!

cc: @colinsauze @unode @mkuzak @amandamiotto @gcapes @joelnitta @vbagadia @jayrobwilliams @chendaniely

**Comment by dsmits** [comment]
Yes I created the slide. Feel free to incorporate it into the lesson material. 
I find it hard to find the source of the monkey cookie cutter images though. I'm pretty sure it was available under CC0 but I didn't credit the creator.

When I reverse image search now I get a couple of results but I don't think they were the original source.

**Comment by jcohen02** [comment]
> Yes I created the slide. Feel free to incorporate it into the lesson material.

That's great, many thanks. 
(And thanks for the information re the images too.)

**Comment by joelnitta** [comment]
- Name: Joel Nitta
- Institution: Chiba University
- Date Taught: [2024-05-30](https://swcarpentry-ja.github.io/2024-05-30-chiba/)
- Was this taught as a stand alone workshop or as part of another workshop / conference?: Part of another workshop
- What feedback do you have on the lesson? Please see issues #217 and #219 and PR #218

- Anything else you'd like to share?

I found that this lesson fit really nicely into the SWC workshop. We taught the first half of `shell-novice`, `git-novice`, then this lesson. I think it works well because all three are foundational tools for reproducible analysis regardless of your programming language of choice. I could imagine then running a more in-depth workshop on (most likely) either R or python for folks who are more interested in either of those.

I really hope this becomes an official part of the SWC curriculum. Please let me know if there's anything I can do to help make that happen!

**Comment by aturner-epcc** [comment]
@joelnitta Thanks or the update and the kind feedback. Glad you found the course useful. We are planning a review of where we are on the move to an official Carpentries lesson and will keep your kind offer of help in mind.

**Comment by colinsauze** [comment]
Name: Colin Sauze

Institution: National Oceanography Centre

Date Taught: 2025-03-21

Was this taught as a stand alone workshop or as part of another workshop / conference? Standalone

What feedback do you have on the lesson? We managed to teach the course in about 5 hours of actual teaching time (with several breaks). Learner feedback was generally positive but some people would have liked more in depth information, some wanted examples using containers in research and others wanted more technical information about how containers work.

Anything else you'd like to share? We suggested Windows users run Podman instead of installing Docker Desktop due to licensing concerns. We had some people who had problems getting Podman running on their Windows systems due to WSL problems and a couple who weren't logged in as an admin user (but did have admin rights). We should have tested Podman on a standard Windows laptop first! We also setup a cloud VM with Podman-docker installed, this worked well for people who couldn't get it working on their machines. The podman-docker package meant we could still type all the commands as "docker ..." but they would really run podman, the only thing which was a little different was that when doing a pull operation it asked which container registry to use and the "docker login" required an additional option of "docker.io" to specify Docker Hub.

**Comment by sstevens2** [comment]
@colinsauze Thanks for your report about your workshop!  When I taught it in fall 2024 we used podman instead too and agreed it was a good replacement except for the setup (we had to solve a lot of install issues at the beginning of the workshop). Cloud VM is interesting solution.  I'd not try the docker-podman package, so glad to hear your feedback on that.  I used podman directly and just tried to remember not to type docker (was only semi-successful lol).  Good note on the login and container registry issue as well.
I'm undecided on what to do with this lesson with the licensing change. I'd like to switch it to podman but we are also still stalled in the middle of review for the lab.  I feel like switching after getting into the lab wouldn't be appropriate but I worry about not honoring the reviewers time on the lesson if we withdraw it from review now and switch and then resubmit.  I need to talk to Toby about what he might suggest and get the maintainers together to discuss how we should move forward.
